admin_manual/installation/nginx.rst

```diff
@@ -157,3 +157,23 @@ If you just see some correct requests in access log, but no login happens, you c
     chown nginx:nginx /var/lib/php/session/
     chown root:nginx /var/lib/php/wsdlcache/
     chown root:nginx /var/lib/php/opcache/
+
+Troubleshooting
+---------------
+
+Service Discovery (aka ``.well-known``)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Command line analysis:
+    To get an example ``curl`` command, check the :doc:`Troubleshooting Guide <../issues/general_troubleshooting>`
+    for tips on how to verify the correct redirection of ``/.well-known`` services by consulting the **Tip** boxes
+    provided in the respective sub-sections of it's :ref:`service-discovery-label` section.
+
+Custom 404 error page:
+    Please note, that using a custom error page for status code ``404`` (along the lines of ``error_page 404 = @error404;``)
+    prevents the http header ``x-nextcloud-well-known`` beeing returned from ``index.php``. As a result, the Overview page
+    in Administration Settings will show a warning, that the endpoints for
+    :ref:`webfinger and hostinfo <service-discovery-webfinger-and-nodeinfo-label>` are not correctly resolved,
+    although in fact they may well be. Please use the above mentioned **Command line analysis** and
```

Comment:
```suggestion
    :ref:`webfinger and hostinfo <service-discovery-webfinger-and-nodeinfo-label>` are not correctly resolved. Please use the above mentioned **Command line analysis** and
```

You cannot use a custom error handler with Nextcloud.

Reply:
Thanks for your suggestion.

The whole point of the test with curl command line is to find out the **real** cause of the problem. In my case, I've spent hours in checking redirection, as the warning message suggests there is something wrong with the redirection. Where in fact, the error resulted because a custom error page was configured in the web server for 404 status code. This custom error page removed the required **header** from the response, which is the actual reason why the test in admin settings overview failed. _I.e. it was not the redirection_.

So with the sentence above, I try to hint to the possibility that the redirection might be working, yet due to the removal of the header, the test fails. The `curl` command line analyses does actually test for that header.

As such, it would be counterproductive to remove the part "although in fact they may well be", as it is not enough to just *resolve* the endpoints correctly.